xbmc/threads/Thread.cpp

```diff
@@ -20,6 +20,7 @@
  *
  */
 
+#include "Application.h"
```

Comment:
I know what you mean, but another solution was not present for me. Problematic is the `main thread` what is normally not present inside this class. Also does the place normally not know (where the exception handling becomes needed) in which thread he is and need to call always something to identify them.

With them is the registration on the class a `static` function where only in case of a segmention fault the thread becomes identified and if on first becomes it done there.

It does nothing direct use something on them, only is for me "CApplication" the Main Thread.